src/main/java/seedu/rc4hdb/logic/commands/modelcommands/FilterCommand.java

```diff
@@ -0,0 +1,174 @@
+package seedu.rc4hdb.logic.commands.modelcommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.rc4hdb.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.rc4hdb.commons.core.Messages;
+import seedu.rc4hdb.commons.util.CollectionUtil;
+import seedu.rc4hdb.logic.commands.CommandResult;
+import seedu.rc4hdb.logic.commands.exceptions.CommandException;
+import seedu.rc4hdb.model.Model;
+import seedu.rc4hdb.model.person.Address;
+import seedu.rc4hdb.model.person.AttributesMatchKeywordsPredicate;
+import seedu.rc4hdb.model.person.Email;
+import seedu.rc4hdb.model.person.Name;
+import seedu.rc4hdb.model.person.Phone;
+import seedu.rc4hdb.model.tag.Tag;
+
+/**
+ * Filters and lists all persons in address book whose attributes are equal to any of the argument keywords.
+ * Keyword matching is case sensitive.
+ */
+public class FilterCommand extends ModelCommand {
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": filters the details of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com";
+
+    public static final String MESSAGE_FILTER_PERSON_SUCCESS = "filtered Person: %1$s";
+    public static final String MESSAGE_NOT_FILTERED = "At least one field to filter must be provided.";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
```

Comment:
Is this really a problem that could be encountered by filtering? Seems more like a problem that would happen when adding a person to the list. Could we possibly remove this?